src/Umbraco.Web.BackOffice/Controllers/MacroRenderingController.cs

```diff
@@ -149,6 +162,21 @@ private async Task<IActionResult> GetMacroResultAsHtml(string? macroAlias, int p
         }
     }
 
+    private static IPublishedContent? GetPagePublishedContent(string pageId, IUmbracoContext umbracoContext)
+    {
+        if (int.TryParse(pageId, NumberStyles.Integer, CultureInfo.InvariantCulture, out int pageIdAsInt))
+        {
+            return umbracoContext.Content?.GetById(true, pageIdAsInt);
+        }
+
+        if (Guid.TryParse(pageId, out Guid pageIdAsGuid))
+        {
+            return umbracoContext.Content?.GetById(true, pageIdAsGuid);
+        }
```

Comment:
@AndyButland would it be better to have `Guid.TryParse` first.. if page id is `Guid` there is no reason trying parsing to `int`.

```
if (Guid.TryParse(pageId, out Guid pageIdAsGuid))
{
    return umbracoContext.Content?.GetById(true, pageIdAsGuid);
} 

if (int.TryParse(pageId, NumberStyles.Integer, CultureInfo.InvariantCulture, out int pageIdAsInt))
{
    return umbracoContext.Content?.GetById(true, pageIdAsInt);
}
```

I guess `int` is mainly for legacy/fallback.

Reply:
Yes, and it will continue to that as well, but it `pageId` is already `Guid` it won't try parsing to `int`.

Reply:
It also never fully supported `Guid` in v13 😅 
https://github.com/umbraco/Umbraco-CMS/pull/15289